### PR TITLE
Added ALBERT Functionality

### PR DESCRIPTION
**Pull request by EricFillion**
**Changes**

- Added ALBERT functionality for TC, QA and WP along with test cases 
- Added BERT test cases for TC, QA and WP
- Added a public variable called "model_type_error" to HappyTransformer. 
- ALBERT for WP: removed "_" from the start of token_str values 

**Testing**
- tests/test_qa.py
- tests/test_tc.py
- tests/test_wp.py

### CODE REVIEW

**Comment by ted537** [comment]
Now that we have 3 separate models, can we codify the lookup processor somewhere to reduce the `if elif` business?

Something like

```
# model_parts.py
# using actual class here because namedtuples defaults support is meh
class ModelParts:
    def __init__(self,tokenizer, seq=None, qa=None, mlm=None, etc):
       self.tokenizer = tokenizer
       # more boring init stuff

MODEL_PARTS = {
    'bert':ModelParts(BertTokenizerFast,seq=BertForSequenceClassification)
}

# usage
from model_parts import MODEL_PARTS
parts = model_parts['BERT']
model = model_parts.seq.from_pretrained(model_name)
```

**Review by ted537 — CHANGES_REQUESTED** [review]
see other comment

**Comment by EricFillion** [comment]
> Now that we have 3 separate models, can we codify the lookup processor somewhere to reduce the `if elif` business?
> 
> Something like
> 
> ```
> # model_parts.py
> # using actual class here because namedtuples defaults support is meh
> class ModelParts:
>     def __init__(self,tokenizer, seq=None, qa=None, mlm=None, etc):
>        self.tokenizer = tokenizer
>        # more boring init stuff
> 
> MODEL_PARTS = {
>     'bert':ModelParts(BertTokenizerFast,seq=BertForSequenceClassification)
> }
> 
> # usage
> from model_parts import MODEL_PARTS
> parts = model_parts['BERT']
> model = model_parts.seq.from_pretrained(model_name)
> ```

We can look into doing something like this when we have more models. For now it's pretty simple and concise just to use if and elif statements, but I agree we should eventually make a transition

File: tests/test_tc.py [hunk 1]
```diff
@@ -73,3 +73,54 @@ def test_qa_train_effectiveness():
...
+
+def test_qa_test_bert():
+    """
+    Tests
+    HappyQuestionAnswering.test()
+    """
+    happy_tc = HappyTextClassification(model_type="BERT", model_name="textattack/bert-base-uncased-SST-2")
+
+    result = happy_tc.test("../data/tc/test.csv")
+    answer = [[{'label': 'LABEL_1', 'score': 0.9995690584182739}], [{'label': 'LABEL_0', 'score': 0.9981549382209778}], [{'label': 'LABEL_0', 'score': 0.9965545535087585}], [{'label': 'LABEL_1', 'score': 0.9978235363960266}]]
```
Comment (ted537) [inline tests/test_tc.py:111]: line breaks are cool

File: tests/test_qa.py [hunk 2]
```diff
@@ -49,3 +49,41 @@ def test_qa_train_effectiveness():
...
+    Ensures that HappyQuestionAnswering.train() results in
+    lowering the loss as determined by HappyQuestionAnswering.eval()
+    """
+
+    happy_qa = HappyQuestionAnswering("BERT", "mrm8488/bert-tiny-5-finetuned-squadv2")
+    before_loss = happy_qa.eval("../data/qa/train-eval.csv")["eval_loss"]
+    happy_qa.train("../data/qa/train-eval.csv")
+    after_loss = happy_qa.eval("../data/qa/train-eval.csv")["eval_loss"]
+
+    assert after_loss < before_loss
```
Comment (ted537) [inline tests/test_qa.py:83]: good test i like

**Review by ted537 — CHANGES_REQUESTED** [review]

File: tests/test_tc.py (hunk 1 above)
Comment (ted537) [reply tests/test_tc.py:111]: By this I mean there are some ridiculously long lines in here. the list at 111 could be spread out vertically

**Review by ted537 — APPROVED** [review]

**Pull request merged**